modules/admin-ui/src/main/webapp/scripts/shared/resources/eventSchedulingResource.js

```diff
@@ -22,7 +22,13 @@
 
 angular.module('adminNg.resources')
 .factory('EventSchedulingResource', ['$resource', 'JsHelper', function ($resource, JsHelper) {
-  var transformResponse = function (data) {
+  var transformResponse = function (data, headers, status) {
+
+        // an event may have no schedule
+        if (status == 404) {
```

Comment:
Maybe status != 200 instead? That would also cover other cases where there is nothing to be parsed, e.g. 403

Reply:
I've chosen 404 intentional since the service returning anything else than 200 or 404 indicates an error in Opencast and assuming that something is wrong it should then cause a JS error. Hiding that would make finding errors hard.